### ISSUE TITLE
 REPL keybindings cannot be changed after #265 fix

### ISSUE THREAD

**Issue by KristofferC** [JuliaLang/julia#19718]
Running the example at: http://docs.julialang.org/en/latest/manual/interacting-with-julia.html#Customizing-keybindings-1 now gives:

```jl
julia> WARNING: Caught an exception in the keymap:
ERROR: MethodError: no method matching (::##1#3)(::Base.LineEdit.MIState, ::Base.REPL.LineEditREPL, ::String)
The applicable method may be too new: running in world age 20393, while current world is 20396.
Closest candidates are:
  #1(::Any, ::Any...) at /home/kristoffer/.juliarc.jl:18 (method too new to be called from this world context.)
```

It should probably be updated to show how to do it in a post #265 world. cc @vtjnash 

(This is after applying https://github.com/JuliaLang/julia/pull/19697)

**Comment by Shade5** [comment]
Can I work on this?

**Comment by yuyichao** [comment]
This is not a doc isue.

**Comment by tkelman** [comment]
Why not? It's the docs that are incorrect here.

**Comment by yuyichao** [comment]
Because it's a REPL bug.

**Comment by vtjnash** [comment]
https://discourse.julialang.org/t/proposal-for-a-first-class-dispatch-wrapper/1127 would help make this not-an-issue.

**Comment by luchr** [comment]
An ugly workaround?

```diff
diff --git a/base/LineEdit.jl b/base/LineEdit.jl
index 2749222..dfc237e 100644
--- a/base/LineEdit.jl
+++ b/base/LineEdit.jl
@@ -734,7 +734,7 @@ end
 keymap_fcn(f::Void, c) = (s, p) -> return :ok
 function keymap_fcn(f::Function, c)
     return function (s, p)
-        r = f(s, p, c)
+        r = eval(Expr(:call,f,s, p, c))
         if isa(r, Symbol)
             return r
         else

```

**Comment by KristofferC** [comment]
Fixed by #19924
